robot/impl/optional_dependencies_test.go

```diff
@@ -344,3 +345,277 @@ func TestOptionalDependencies(t *testing.T) {
 		test.That(t, oc.optionalMotor, test.ShouldNotBeNil)
 	}
 }
+
+func TestModularOptionalDependencies(t *testing.T) {
+	// A copy of TestNonModularOptionalDependencies with a modular component instead of a
+	// resource defined in this file.
```

Comment:
See comment: pretty much identical test as the existing one but with a modular resource (`foo` from `optionaldepsmodule`).